File: js/src/bindings/registerWithNft.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.registerWithNft = void 0;
const spl_token_1 = require("@solana/spl-token");
const web3_js_1 = require("@solana/web3.js");
const createWithNftInstruction_1 = require("../instructions/createWithNftInstruction");
const constants_1 = require("../constants");
const registerWithNft = (name, space, nameAccount, reverseLookupAccount, buyer, nftSource, nftMetadata, nftMint, masterEdition) => {
    const [state] = web3_js_1.PublicKey.findProgramAddressSync([nameAccount.toBuffer()], constants_1.REGISTER_PROGRAM_ID);
    const ix = new createWithNftInstruction_1.createWithNftInstruction({ space, name }).getInstruction(constants_1.REGISTER_PROGRAM_ID, constants_1.NAME_PROGRAM_ID, constants_1.ROOT_DOMAIN_ACCOUNT, nameAccount, reverseLookupAccount, web3_js_1.SystemProgram.programId, constants_1.REVERSE_LOOKUP_CLASS, buyer, nftSource, nftMetadata, nftMint, masterEdition, constants_1.WOLVES_COLLECTION_METADATA, spl_token_1.TOKEN_PROGRAM_ID, web3_js_1.SYSVAR_RENT_PUBKEY, state, constants_1.METAPLEX_ID);
    return ix;
};
exports.registerWithNft = registerWithNft;

File: python/src/bindings/register_with_nft.py
from solana.constants import SYSTEM_PROGRAM_ID
from solders.instruction import Instruction
from solders.pubkey import Pubkey
from spl.token.constants import TOKEN_PROGRAM_ID

from constants import REGISTER_PROGRAM_ID, NAME_PROGRAM_ID, ROOT_DOMAIN_ACCOUNT, REVERSE_LOOKUP_CLASS, \
    WOLVES_COLLECTION_METADATA, METAPLEX_ID, SYSVAR_RENT_PUBKEY
from instructions.CreateWithNftInstruction import CreateWithNftInstruction


async def register_with_nft(
        name: str,
        space: int,
        name_account: Pubkey,
        reverse_lookup_account: Pubkey,
        buyer: Pubkey,
        nft_source: Pubkey,
        nft_metadata: Pubkey,
        nft_mint: Pubkey,
        master_edition: Pubkey,
) -> Instruction:
    state, _ = Pubkey.find_program_address(
        [bytes(name_account)],
        REGISTER_PROGRAM_ID
    )

    return CreateWithNftInstruction(name, space).get_instruction(
        REGISTER_PROGRAM_ID,
        NAME_PROGRAM_ID,
        ROOT_DOMAIN_ACCOUNT,
        name_account,
        reverse_lookup_account,
        SYSTEM_PROGRAM_ID,
        REVERSE_LOOKUP_CLASS,
        buyer,
        nft_source,
        nft_metadata,
        nft_mint,
        master_edition,
        WOLVES_COLLECTION_METADATA,
        TOKEN_PROGRAM_ID,
        SYSVAR_RENT_PUBKEY,
        state,
        METAPLEX_ID
    )